spyder/plugins/editor/plugin.py

```diff
@@ -2219,6 +2219,11 @@ def debug_command(self, command):
         focus_widget = self.main.ipyconsole.get_focus_widget()
         if focus_widget:
             focus_widget.setFocus()
+        current_editor = self.get_current_editor()
+        if current_editor is not None:
+            current_sw = self.main.ipyconsole.get_current_shellwidget()
+            if current_sw is not None and current_sw._reading:
+                current_editor.sig_stop_debugging.emit()

```

Comment:
I don't understand the logic of this change. Could you explain it, please?

Reply:
It was a try to use the `_reading` attribute to determine if the console was debugging. Although, after testing it more throughfully it was not working as expected. I did a new implementation using the `sig_prompt_ready` to trigger the handle of the debugging state in the panel